### PR TITLE
Add golden tests infrastructure, and golden tests for the home page

### PR DESCRIPTION
**Pull request by perclasson**
- Add golden tests infrastructure, and golden tests for the home page. Further tests for the most important pages will be added in a follow up.
- Split up the Github Actions yaml files into three different, as the file was getting hard to read. Now we have one for unit tests (`unit.yml`), one for integration tests (`integration.yml`) and one for golden tests (`golden.yml`).

Relates to https://github.com/flutter/gallery/issues/39

### CODE REVIEW

File: .github/workflows/integration.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
+name: Flutter Gallery Integration Tests
```
Comment (rami-a) [inline .github/workflows/integration.yml:1]: This seems unrelated to golden tests, can you update the PR description to include this too (or split into a separate PR if you want)

File: golden_test/home_test.dart [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+  'assets/studies/shrine_card.png',
+  'assets/studies/shrine_card_dark.png',
+  'assets/studies/fortnightly_card.png',
+  'assets/studies/fortnightly_card_dark.png',
+];
+
+void main() {
+  group('mobile', () {
+    testWidgets('home page matches golden screenshot', (tester) async {
+      await setUpBinding(tester);
```
Comment (rami-a) [inline golden_test/home_test.dart:31]: you might want to undo any bindings you set at the end of each test, to avoid any accidental issues. You can do that in a tearDown method instead of inside each test

File: pubspec.yaml [hunk 3]
```diff
@@ -114,4 +114,3 @@ flutter:
     - family: GalleryIcons
       fonts:
         - asset: fonts/GalleryIcons.ttf
-
```
Comment (rami-a) [inline pubspec.yaml:117]: Undo this change.

File: golden_test/README.md [hunk 4]
```diff
@@ -0,0 +1,22 @@
+# Golden tests
+Golden tests, also called screenshot tests, let's you write tests to easily find UI regressions. It
+compares the UI to saved goldens screenshots, and if there is a pixel difference the tests will
+fail. It gives us confidence that code changes does not cause UI regressions.
+
+## Updating goldens
+An expected change to the UI will also cause the golden tests to fail. You will then need to update
```
Comment (willlarche) [inline golden_test/README.md:7]:
```suggestion
An expected change to the UI will also cause the golden tests to fail. If this is unintended, you will need to change your code to prevent the UI change. If it was intended, you will then need to update
```

File: golden_test/README.md [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+Due to slight [differences](https://github.com/flutter/flutter/issues/36667#issuecomment-521335243)
+in rendering across platforms, mostly around text, the tests will only be run on a macOS machine
+on Github Actions. This means that if you update the tests on a Linux or Windows machine the golden
+tests will not pass on Github Actions. Instead you are recommended to download the goldens directly
+from the failed Github Actions job, and use those inside of your branch.
+
+You can find the goldens under Artifacts in the top menu inside of the failed Github Actions job. It
+will include the master image, your test image and the difference.
+
+![Where to download the golden artifacts](https://user-images.githubusercontent.com/1770678/80202787-70f1de80-8626-11ea-9d98-58ac72f67479.png)
```
Comment (willlarche) [inline golden_test/README.md:22]: Add a section explaining why Shadows are opaqued.

Comment (willlarche) [inline golden_test/README.md:22]: Add a section explaining why we have to do font loading.

File: .github/workflows/integration.yml (hunk 1 above)
Comment (perclasson) [reply .github/workflows/integration.yml:1]: Updated the PR description. I just moved some content to another file.

File: golden_test/README.md (hunk 4 above)
Comment (perclasson) [reply golden_test/README.md:7]: Done.

File: golden_test/README.md (hunk 5 above)
Comment (perclasson) [reply golden_test/README.md:22]: Done!

Comment (perclasson) [reply golden_test/README.md]: Done.

File: golden_test/home_test.dart (hunk 2 above)
Comment (perclasson) [reply golden_test/home_test.dart:31]: Added a tear down now to clear the test values:

`addTearDown(tester.binding.window.clearAllTestValues);`
